script/testing/util/common.py

```diff
@@ -22,55 +26,83 @@ def run_command(command,
+        lines = file.readlines()
+        for line in lines:
+            LOG.info(line.strip())
+
+
+def format_time(timestamp):
+    return datetime.datetime.fromtimestamp(timestamp).strftime(
+        "%Y-%m-%d %H:%M:%S")
+
+
+def kill_pids_on_port(port):
+    for proc in psutil.process_iter():
+        try:
+            for conns in proc.connections(kind="inet"):
+                if conns.laddr.port == port:
+                    LOG.info(
+                        "Killing existing server instance listening on port {} [PID={}], created at {}"
+                        .format(port, proc.pid,
+                                format_time(proc.create_time())))
+                    proc.send_signal(signal.SIGKILL)
+        except psutil.ZombieProcess:
+            LOG.info("Killing zombie process [PID={}]".format(proc.pid))
+            proc.parent().send_signal(signal.SIGCHLD)
+
+
+def get_pids_on_port(port):
     """Get the list of PIDs (if any) listening on the target port"""
-    
-    # Copied from https://gist.github.com/jossef/593ade757881bb7ddfe0
-    # I would like to use psutil to make this more portable but that would require
-    # us to install an additional package with pip
-    
-    command = "lsof -i :%s | awk '{print $2}'" % port
-    output = subprocess.check_output(command, shell=True).strip()
-    if output:
-        output = re.sub(' +', ' ', output.decode('utf-8'))
-        for pid in output.split('\n'):
-            try:
-                yield int(pid)
-            except:
-                pass
-
-
-def check_pid(pid):
-    """Check whether pid exists in the current process table."""
-    
-    # Copied from psutil
-    # https://github.com/giampaolo/psutil/blob/5ba055a8e514698058589d3b615d408767a6e330/psutil/_psposix.py#L28-L53
-    
-    if pid == 0:
-        return True
-    try:
-        os.kill(pid, 0)
-    except OSError as err:
-        if err.errno == errno.ESRCH:
-            # ESRCH == No such process
-            return False
-        elif err.errno == errno.EPERM:
-            # EPERM clearly means there's a process to deny access to
-            return True
-        else:
-            # According to "man 2 kill" possible error values are
-            # (EINVAL, EPERM, ESRCH) therefore we should never get
-            # here. If we do let's be explicit in considering this
-            # an error.
-            raise err
-    else:
-        return True
+
+    pids = []
+    for proc in psutil.process_iter():
+        try:
+            for conns in proc.connections(kind="inet"):
+                if conns.laddr.port == port:
+                    pids.append(proc.pid)
+        except psutil.ZombieProcess:
+            # ignore the zombie process
+            continue
+    return pids
+
+
+def check_pid_exists(pid):
+    """ Checks to see if the pid exists """
+    return psutil.pid_exists(pid)
+
+
+def run_check_pid_exists(pid):
+    cmd = "sudo python3 {SCRIPT} {PID}".format(SCRIPT=FILE_CHECK_PID_EXISTS,
```

Comment:
You need documentation that explains why we have to fork with sudo to run this function.

Reply:
function docstrings added for both `run_check_pid_exists` and `run_kill_pids_on_port`